compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -198,6 +198,8 @@ where
         if ty.has_escaping_bound_vars() {
             let (ty, mapped_regions, mapped_types, mapped_consts) =
                 BoundVarReplacer::replace_bound_vars(infcx, &mut self.universes, ty);
+            // FIXME: Temporary placeholders may get added to infcx's region constraints/obligations,
+            // which can cause problem for `resolve_regions`.
```

Comment:
what do you mean here? I agree that we may get constraints involving placeholders from this, but that's fine™

We have the same behavior when relating binders and what not. The way we handle such placeholder constraints may change going forward and ideally we'd eagerly handle all of them when "exiting" the binder again, but for now "leaking" these constraints is *normal*